crates/cli/src/git.rs

```diff
@@ -9,6 +9,27 @@ pub struct Origin {
     pub url: String,
 }
 
+pub async fn git_changed_files() -> Result<Vec<String>> {
+    let output = Command::new("git")
+        .args(["status", "--ignored=no", "--short"])
+        .output()
+        .await?;
+
+    if !output.status.success() {
+        return Err(anyhow::anyhow!(
+            "Failed to get git changed files: {}",
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    };
+
+    let changed_files = String::from_utf8(output.stdout)?
+        .lines()
+        .map(|row| row.trim().split(' ').last().unwrap().to_string())
+        .collect();
+
+    Ok(changed_files)
+}
+
```

Comment:
@grant0417 is this CLI run on the backend too? If the answer is affirmative this fn won't work as it tries to detect uncommited changes. It should work correctly in local environment however, I am testing it later.

Reply:
Yeah, this run on the backend as well. We might need to pass more info do you can diff between the from and to branch